lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -575,7 +575,8 @@ sub is_ok ($self) {
 sub is_ok_to_retry ($self) {
     return 1 unless my $result = $self->result;
     return 0 if grep { $_ eq $result } OK_RESULTS;    # retry is not needed if job is ok
-    return 0 if $result eq USER_CANCELLED;    # retry is not needed if job is user-cancelled
+    return 0 if $result eq USER_CANCELLED;    # retry is not needed if job is user-cancelled...
+    return 0 if $result eq OBSOLETED;    # ...or obsoleted
```

Comment:
The only consideration is whether this statement should come before the `USER_CANCELLED`. probably it doesnt make big difference though. Just as a sequence I think it is more intuitive. Other than that, LGTM

Reply:
i really don't think it matters. in *theory* the one most likely to be encountered should come first, I guess. I could have made them into a two-item array and changed the check into a `grep`, but it just didn't feel worth it.

Reply:
Did you consider to use `ABORTED_RESULTS` from lib/OpenQA/Jobs/Constants.pm? That would be

```
use constant ABORTED_RESULTS =>
  (SKIPPED, OBSOLETED, PARALLEL_FAILED, PARALLEL_RESTARTED, USER_CANCELLED, USER_RESTARTED);
```

I would assume in all those cases we can also skip the retry. @Martchus WDYT?

Reply:
I specifically decided against using it because of `PARALLEL_FAILED`; we definitely want retries in that case (if job A and job B are parallel, and A fails, we definitely want B to be retried). But if, as you say, such a job actually would still be retried in this case, we can reconsider it. I think we should investigate exactly when each of those states can happen first, though. Sometimes words lie. In any case, it seems like a rather more significant change than this one (I already checked that `OBSOLETED` really is *only* used when a job is eaten by a new schedule request).

Reply:
fine, your choice is the safe one

Reply:
I dared to create
* #5806